build-flannel-resources.sh

```diff
@@ -1,8 +1,8 @@
 #!/usr/bin/env bash
 set -eux
 
-FLANNEL_VERSION=${FLANNEL_VERSION:-"v0.11.0"}
-ETCD_VERSION=${ETCD_VERSION:-"v2.3.7"}
+FLANNEL_VERSION=${FLANNEL_VERSION:-"v0.20.2"}
+ETCD_VERSION=${ETCD_VERSION:-"v3.4.22"}
```

Comment:
I was going to try [uploading the new Flannel resources](https://github.com/charmed-kubernetes/charm-flannel/pull/85#issuecomment-1410491501) to my cluster, but I think if I only run `juju attach-resource flannel …` with the latest `flannel.tar.gz` then this etcd upgrade will break my Flannel charm.

My thinking is:

1. The latest Flannel resource archives contains a major version bump of etcd (from 2, in my current charm, to 3).
2. My charm will _not_ have the changes to `reactive/flannel.py` made in this PR.
3. The new `etcd` binary assumes API version 3 by default.
4. The etcd API assumed in _my charm's_ `reactive/flannel.py` is version 2, which is incompatible with version 3.
5. My flannel charm will then fail in talking to etcd.

Does that sound right to you @addyess?

Reply:
Indeed, the cluster is using the canal charm.

That sounds great, I will update the `canal` charm once the new version is out 🚀 

Thanks again for all the help!

Reply:
it's POSSIBLE to upgrade to the latest/edge -- but do so assuming this has yet to fully go through quality testing.

```bash
juju upgrade-charm canal --channel latest/edge
```

Reply:
We tried this on our cluster a few weeks ago. The upgrade completed successfully and we haven't seen the problem mentioned in #85 since the upgrade, so I'd call that a victory!

Thanks again for all your help! 🏅

Reply:
@alexpearce thank you so so much for your contribution here.